benchopt/plotting/plot_histogram.py

```diff
@@ -40,27 +43,39 @@ def plot_histogram(df):
         df_tol = df_.groupby('stop_val').filter(
             lambda x: x['objective_value'].max() < c_star)
         if df_tol.empty:
+            colors[i] = "w" if not plotly else (.8627, .8627, .8627)
             print(f"Solver {solver_name} did not reach precision {eps}.")
-            height = df.time.max()
-            rect = ax.bar(
-                x=xi, height=height, width=width, color='w', edgecolor='k')
-            ax.annotate("Did not converge", xy=(xi, height/2), ha='center',
-                        va='center', color='k', rotation=90)
-            rect_list.append(rect)
+            height_list.append(df.time.max())
+            times_list.append(np.nan)
             continue
         stop_val = df_tol['stop_val'].min()
         this_df = df_[df_['stop_val'] == stop_val]
-        rect_list.append(ax.bar(
-            x=xi, height=this_df['time'].mean(), width=width, color=colors[i]))
+        height_list.append(this_df['time'].mean())
```

Comment:
I think it is safer to use the median time instead of the average one as it is more robust to large deviations caused for instance by overloaded systems:
```suggestion
        height_list.append(this_df['time'].median())
```